### ISSUE TITLE
Should you be able to scale the marks available for questions in an exam?

### ISSUE THREAD

**Issue by christianp** [numbas/Numbas#476]
We have a situation where we want to reuse questions from a variety of existing exercises, but we want to reweight the questions. At the moment, the only way of doing this is to make copies of the questions.

Should it be possible to say, at the exam level, how many points each question is worth? The marks available for each part would then be rescaled.

One potential drawback is that you'd get a question where each part is worth something like `6/7 = 0.857143` marks. Even if we only show marks to 2 d.p., so 0.86 in this case, would that look odd?

**Comment by WoosterUK** [comment]
Yes! This is one of my constant irritations: we have a computer to do the marking, so it can do the adding up, too. We should be separating the question of how many marks it is sensible to assign to a question from how significant the question is in the context of the assessment as a whole.

So perhaps use percentages with percentage weightings? For example (I can't get a table to work in this box):

Q1: weight 33%, points 7/11, grade 21.0%.
Q2: weight 20%, points 3/4, grade 15.0%.
Q3: weight 47%, points 5/9, grade 26.1%.
Total: grade 62.1%.

**Comment by christianp** [comment]
Many users will be much less comfortable with odd-looking percentages than you are.

There's also the problem of how we switch to this system - since the marks available can't be fully known until runtime, thanks to randomised multiple choice parts, it's not straightforward to assign weightings to existing exams that preserve the old score distributions.

**Comment by christianp** [comment]
This came up again while planning the open resource library. I'm now convinced that we should be able to scale each question at the exam level, so users don't have to make copies of questions they're reusing.

**Comment by zaript** [comment]
Perhaps, this could use the same interface as exam level variables override?

Also, I find it more convenient to assign marks to questions, rather than weights, because the weights need to be readjusted when questions added or removed.

My suggestion is to add "exam_level_max_mark" parameter and scale all the question marks by "exam_level_max_mark"/"question_level_max_mark".
